src/main/java/net/minecraftforge/common/crafting/conditions/ICondition.java

```diff
@@ -5,11 +5,33 @@
 
 package net.minecraftforge.common.crafting.conditions;
 
+import net.minecraft.core.Holder;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.Tag;
+
+import javax.annotation.Nullable;
+import java.util.Map;
 
 public interface ICondition
 {
     ResourceLocation getID();
 
+    default boolean test(@Nullable IContext context)
```

Comment:
What about if instead of making this nullable, you create `IContext.EMPTY` as a default to pass in? Saves having to do a bunch of extra null checks in conditions.

Reply:
Good idea, implemented!